src/runtime/ranger/ranger_resource_policy_manager.cpp

```diff
@@ -0,0 +1,560 @@
+
+    resource.insert(std::make_pair(code, type));
+}
+
+bool ranger_resource_policy_manager::allowed(const int rpc_code,
+                                             const std::string &user_name,
+                                             const std::string &database_name)
+{
+    {
+        utils::auto_read_lock l(_global_policies_lock);
+        if (_rpc_match_global_acl.find(rpc_code) != _rpc_match_global_acl.end()) {
+            for (auto &item : _global_policies) {
+                const access_type &type = _rpc_match_global_acl[rpc_code];
+                if (item._policies.allowed(user_name, type)) {
+                    return true;
+                }
+            }
+            if (rpc_code != RPC_CM_LIST_APPS.code()) {
+                return false;
+            }
+        }
+    }
+    {
+        utils::auto_read_lock l(_database_policies_lock);
+        if (_rpc_match_database_acl.find(rpc_code) != _rpc_match_database_acl.end()) {
+            for (auto &item : _database_policies) {
+                const access_type &type = _rpc_match_database_acl[rpc_code];
+                if (item._policies.allowed(user_name, type)) {
+                    if ("" == database_name &&
+                        item._database_values.find("*") != item._database_values.end()) {
+                        return true;
+                    }
+                    if (item._database_values.find(database_name) != item._database_values.end()) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+    }
+    // old env deal way: todo
+    return false;
+}
+
+void ranger_resource_policy_manager::update()
+{
+    dsn::error_code err_code = load_ranger_resource_policy();
+    if (err_code == dsn::ERR_RANGER_POLICIES_NO_NEED_UPDATE) {
+        LOG_DEBUG("No need to update ACLs policies with error code = {}", err_code);
+        return;
+    }
+    if (err_code != dsn::ERR_OK) {
+        LOG_WARNING("get ranger policy failed, {}.", err_code);
+    } else {
+        LOG_DEBUG("get ranger policy success, {}.", err_code);
+        dsn::task_ptr after_create_ranger_policy_meta_root = dsn::tasking::create_task(
+            LPC_CM_GET_RANGER_POLICY, &_tracker, [this]() { start_sync_ranger_policies(); });
+        create_ranger_policy_root(after_create_ranger_policy_meta_root);
+    }
+}
+
+void ranger_resource_policy_manager::create_ranger_policy_root(dsn::task_ptr callback)
+{
+    LOG_DEBUG("create ranger policy meta root({}) on remote_storage",
+              _ranger_policy_meta_root.c_str());
+    _meta_svc->get_remote_storage()->create_node(
+        _ranger_policy_meta_root, LPC_CM_GET_RANGER_POLICY, [this, callback](dsn::error_code err) {
+            if (err == dsn::ERR_OK || err == ERR_NODE_ALREADY_EXIST) {
+                LOG_DEBUG("create ranger policy meta root({}) succeed, with err ({})",
+                          _ranger_policy_meta_root.c_str(),
```

Comment:
```suggestion
                          _ranger_policy_meta_root,
```
日志风格再自己梳理一遍吧，包括LOG、CHECK

Reply:
done